cbor/record.go

```diff
@@ -20,31 +20,45 @@ func init() {
 
 // record defines the node structure of a record.
 type record struct {
-	Block cid.Cid
-	Sig   []byte
-	Prev  cid.Cid `refmt:",omitempty"`
+	Block     cid.Cid
+	Sig       []byte
+	AuthorSig []byte
```

Comment:
Here's the new signature. When a node receives a record, it should be able to independently verify who wrote it, which may not be the sender, e.g., on an initial thread join, where all logs and records are pushed.

Reply:
So, `Sig` is the signature using the Log private key.
And `AuthorSig` is the signature by the owner of the Log?

Not clear yet why both things are necessary at the `record` layer, since whoever owns the private keys of the Log is, de facto, the author. (Or not, and I'm confused).

Reply:
Maybe it wasn't clear in the description, the author sig will be used to check against the thread ACL. My thinking is that the ACL will reference author pub keys, not log pub keys, since an author can have many logs. The ACL implementation (and verifications) are not in the PR.

Reply:
Oh, I see.
A next question is: it make sense that a Log is owned by more than one author?

Because if a log is only owned by one author, it seems that the ACL check should be against the Log (which indirectly has the owner, but maybe not stored here).

If a log can be written by more than one author, it seems that isn't a single-writer log?

Reply:
Well, that _not stored here_ thing might be the tricky part. Where if not here it can live. Not completely sure. Have you thought any other option than this one?

Reply:
Take the case of an ETH or 3Box key, or whatever other key you want to think of as _your identity_, these would be the author key. So, you'd have one log per author per thread, which is single writer.

Do you mean where else could the author sig live? Originally, I had thought it was only needed in the over-the-wire messages (pb messages), but then realized that sometimes you get records from the non-author, e.g., when you're getting pushed a full log from a replicator.

Reply:
Yeah, sure... I can receive logs from the author, or some other peer that is sharing the info.

But for ACL verification, I care about that distinction? Maybe that's the point I'm missing.
If the ACL says _Author1 can't write to the thread_ , I can apply that rule not caring about if I received a new record from Author1 owned-log from himself directly or from a sharing peer.

No matter how sends me a new record from a Log owned by Author1, I'll deny it.

Reply:
To be clear, I'm forcing some questions just to double-check if make sense to you. I think for me will click when the ACL impl lands, and see the pieces together. So feel free to dismiss a bit my noise.

Reply:
Right, that's why it the signature should be in the record, and not in the over-the-wire message... sorry, that comment above may have been confusing, I was trying to address your comment about where to store the signature. Though, I may have misunderstood the question. 

Another reason author key != log key, is because log are created as needed when a peer joins. But the static ACL is created when the thread is created... at which time, you may not know all the log keys.

Reply:
The noise is good! I ran myself in a couple circle with all this stuff... it's good to chat about it and get push back, so the whole thing comes out right!

Reply:
Cool, we can continue this chat on another place. Will ✔️

Reply:
If you implement the Credentials interface, you could handling the signing however you want... use a local key, call to another service, etc. Both ways are possible, unless I'm missing something.

Reply:
Ooooh, ok fuck yes I hadn't thought about that. We (Textile) could provide a nice wrapper class that'll do this. That opens up sooo many possibilities!